view/statistics-flow-certificates.js

```diff
@@ -0,0 +1,303 @@
+		serviceToCertLegacyMatch[''].push('certificate-' + certificate.key);
+	});
+});
+
+var buildResultRow = function (rowData) {
+	return assign({
+		submitted: 0,
+		pending: 0,
+		pickup: 0,
+		withdrawn: 0,
+		rejected: 0,
+		sentBack: 0
+	}, rowData || {});
+};
+
+var accumulateResultRows = function (rows) {
+	var result = buildResultRow(rows[0]);
+	rows.slice(1).forEach(function (row) {
+		Object.keys(row).forEach(function (propertyName) {
+			result[propertyName] += row[propertyName];
+		});
+	});
+
+	return result;
+};
+
+var filterData = function (data) {
+	var result = {};
+	location.query.get('service').map(function (service) {
+		location.query.get('certificate').map(function (certificate) {
+			Object.keys(data).forEach(function (key) {
+				if (service && certificate) {
+					result[key] = buildResultRow(data[key][service].certificate[certificate]);
+				} else if (service) {
+					result[key] = buildResultRow(data[key][service].businessProcess);
+				} else {
+					var rowsToAccumulate = [];
+					Object.keys(data[key]).forEach(function (service) {
+						if (certificate) {
+							if (!data[key][service].certificate[certificate]) return;
+							rowsToAccumulate.push(data[key][service].certificate[certificate]);
+						} else {
+							rowsToAccumulate.push(data[key][service].businessProcess);
+						}
+					});
+					result[key] = accumulateResultRows(rowsToAccumulate);
+				}
+			});
+		});
+	});
+
+	return result;
+};
+
+var incrementDateByTimeUnit = function (date, mode) {
+	switch (mode) {
+	case 'weekly':
+		date.setUTCDate(date.getUTCDate() + 7);
+		break;
+	case 'monthly':
+		date.setUTCMonth(date.getUTCMonth() + 1);
+		break;
+	case 'yearly':
+		date.setUTCFullYear(date.getUTCFullYear() + 1);
+		break;
+	default:
+		date.setUTCDate(date.getUTCDate() + 1);
+		break;
+	}
+
+	return date;
+};
+
+var calculateDurationByMode = function (dateFrom, dateTo, mode) {
+	var timeUnitsCount = 0, currentDate;
+	currentDate = new db.Date(dateFrom.getTime());
+	if (!dateTo) dateTo = new db.Date();
+
+	while (currentDate < dateTo && timeUnitsCount < 600) {
+		timeUnitsCount++;
+		incrementDateByTimeUnit(currentDate, mode);
+	}
+	return timeUnitsCount;
+};
+
+exports['statistics-main'] = function () {
+	var queryHandler, data = new ObservableValue({}), pagination = new Pagination('/flow/');
+	queryHandler = setupQueryHandler(getQueryHandlerConf(), location, '/flow/');
+	queryHandler.on('query', function (query) {
+		var serverQuery = copy(query), dateFrom, dateTo, mode
+		  , currentDate, offset, timeUnitsCount = 0, durationInTimeUnits;
+
+		dateFrom = new db.Date(location.query.get('dateFrom').value);
+		dateTo = query.dateTo || new db.Date();
+		mode = query.mode;
+		durationInTimeUnits = calculateDurationByMode(dateFrom, dateTo, mode);
+		pagination.count.value = Math.ceil(durationInTimeUnits / itemsPerPage);
+		pagination.current.value = Number(location.query.get('page').value) || 1;
+		if (pagination.count.value > 1) {
```

Comment:
- `dateTo`, `dateFrom` should already be final on `query` (no other processing of those values should be needed, apart of serializing them to ISO string before sending them with query to the server)
- Paginator interface should not be changed at this moment (we should do it when we fill table with data, line 178).
- `page` resolution should be made in queryHandler (as it happens in other paginated tables, e.g. business processes table). It's important that `queryHandler` validates complete query and not part of it (mind it also fixes the query in address bar if it's invalid, you don't want to pursue this things manually aside).

You can get notion of what handling is expected by investigating other configurations. e.g. in what we do in case of `inspector`:
  1. Query handler normalizes and validates query, after that's done it emits `query` event -> https://github.com/egovernment/eregistrations/blob/master/view/components/inspector-table/setup-query-handler.js#L14
  2. That runs `manager.update` which just does this -> https://github.com/egovernment/eregistrations/blob/master/view/components/objects-table-simple/manager.js#L46-L51. This is about some internal caching, and not really needed (at least at this point) normalization of `query.page` to number. `query` is passed to `queryExternal`.
  3. `queryExternal` directly passes query to server -> https://github.com/egovernment/eregistrations/blob/master/view/components/inspector-table/manager.js#L19-L20

Best if this configuration follows same convention.

The one tricky part, (but nothing that can't be cleanly solved), It's fact that for _paginator interface update_ (which should be done after we receive result from server) we need to know total number of pages, which is also internally needed by _query handler_ to properly resolve `page`.
I see few solutions:
- (better one) Introduce generic util that calculates `pagesCount` based on query, and for one update invoke it twice (1) internally in `queryHandler` and (2) when want to update paginator interface.
It seems not perfectly clean when we think of algorithm (same calculation runs twice), but we can easily fix that with memoization trick (this util will be run twice with same query object) in that case we can also rely on [weak map based memoization](https://github.com/medikoo/memoizee#weakmap-based-configurations) so we're friendly for garbage collector
- As we do not want to pass exactly same `query` to server, we can include `pagesCount` calculation logic purely in query handler, and from it expose both `page` and `pageCount` on resolved `query` (in query conf functions `ensure` receives `query` as second argument, so we can add `pageCount` in `page` handler).. Then from this `query` we can read values for paginator